src/frontend-samples/global-display-sample/GlobalDisplayUI.tsx

```diff
@@ -0,0 +1,188 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { BackgroundMapType } from "@bentley/imodeljs-common";
+import { IModelApp, IModelConnection, ScreenViewport, SpatialViewState } from "@bentley/imodeljs-frontend";
+import "common/samples-common.scss";
+import { ControlPane } from "common/ControlPane/ControlPane";
+import { SandboxViewport } from "common/SandboxViewport/SandboxViewport";
+import * as React from "react";
+import { Button, Input, Toggle } from "@bentley/ui-core";
+import GlobalDisplayApp from "./GlobalDisplayApp";
+
+interface GlobalDisplayUIState {
+  viewport?: ScreenViewport;
+  /** Place name to which to travel. */
+  destination: string;
+  /** True for 3d terrain, false for a flat map. */
+  terrain: boolean;
+  /** Display map labels with the map imagery. */
+  mapLabels: boolean;
+  /** Display 3d building meshes from Open Street Map Buildings. */
+  buildings: boolean;
+  /** If buildings are displayed, also display their edges. */
+  buildingEdges: boolean;
+}
+
+interface GlobalDisplayUIProps {
+  iModelSelector: React.ReactNode;
+  iModelName: string;
+}
+
+export default class GlobalDisplayUI extends React.Component<GlobalDisplayUIProps, GlobalDisplayUIState> {
+  public state: GlobalDisplayUIState = {
+    destination: "",
+    terrain: false,
+    buildings: false,
+    buildingEdges: false,
+    mapLabels: true,
+  };
+
+  private readonly onIModelReady = () => {
+    IModelApp.viewManager.onViewOpen.addOnce((viewport: ScreenViewport) => {
+      // The grid just gets in the way - turn it off.
+      viewport.view.viewFlags.grid = false;
+
+      // We're not interested in seeing the contents of the iModel, only the global data.
+      if (viewport.view.isSpatialView())
+        viewport.view.modelSelector.models.clear();
+
+      this.setState({ viewport, terrain: true, buildings: true, buildingEdges: true, mapLabels: false });
+    });
+  };
+
+  private readonly travelToDestination = async () => {
+    if (this.state.viewport)
+      GlobalDisplayApp.travelTo(this.state.viewport, this.state.destination);
```

Comment:
Would be nice to get some feedback for when no location is found. Just does nothing at the moment.

Reply:
You guys don't implement most of the UI stuff (toasts, popups, etc). What kind of feedback do you want?

Reply:
For example, this attempt executes but produces no visible message:
```
  private readonly travelToDestination = async () => {
    if (!this.state.viewport)
      return;

    const locationFound = await GlobalDisplayApp.travelTo(this.state.viewport, this.state.destination);
    if (!locationFound) {
      const message = `Sorry, "${this.state.destination}" isn't recognized as a location.`;
      IModelApp.notifications.outputMessage(new NotifyMessageDetails(OutputMessagePriority.Warning, message));
    }
```

Reply:
[IOTAlert Sample uses NotificationRenderer](https://github.com/imodeljs/frontend-sample-showcase/blob/master/src/frontend-samples/iot-alert-sample/IotAlertUI.tsx#L195) from ui-framework. Is that not correct?

Reply:
I think toast or popup would be good

Reply:
I don't know, I just know that I can never get you to show a toast message or a popup toolbar.

Reply:
Added MessageRenderer to SampleShowcase